aws_kms_pkcs11.cpp

```diff
@@ -401,20 +401,45 @@ CK_RV C_GetMechanismInfo(CK_SLOT_ID slotID, CK_MECHANISM_TYPE type, CK_MECHANISM
     if (pInfo == NULL) {
         return CKR_ARGUMENTS_BAD;
     }
+    AwsKmsSlot& slot = slots->at(slotID);
+    CK_ULONG keySize = 0;
     switch (type) {
     case CKM_RSA_PKCS:
-        pInfo->ulMinKeySize = 2048;
-        pInfo->ulMaxKeySize = 2048;
-        pInfo->flags = CKF_SIGN;
-        break;
+        switch(slot.GetKeySpec()) {
+        case Aws::KMS::Model::KeySpec::RSA_2048:
+            keySize = 2048;
+            break;
+        case Aws::KMS::Model::KeySpec::RSA_3072:
+            keySize = 3072;
+            break;
+        case Aws::KMS::Model::KeySpec::RSA_4096:
+            keySize = 4096;
+            break;
+        default:
+            // invalid combination of mechanism and KMS key spec
+            return CKR_MECHANISM_INVALID;
+        }
```

Comment:
Still need a break here I believe.
```suggestion
        }
        break;
```

Reply:
Ah, yep 🤦 . Thanks!